tests/functional/monitoring/prometheus/alerts/test_alerting_works.py

```diff
@@ -39,17 +40,48 @@ def test_alerting_works(threading_lock):
 def test_prometheus_rule_failures(threading_lock):
     """
     There should be no PrometheusRuleFailures alert when OCS is configured.
+    This test is extended to check for many-to-many matching errors in Prometheus logs.
+    If such error message found PrometheusRuleFailures alert must fire as well.
     """
+    # any check with state False will fail the test
+    test_results = {}
+
     prometheus = ocs_ci.utility.prometheus.PrometheusAPI(threading_lock=threading_lock)
     alerts_response = prometheus.get(
         "alerts", payload={"silenced": False, "inhibited": False}
     )
-    assert alerts_response.ok is True
+    test_results["alert-msg-ok-check"] = alerts_response.ok is True
     alerts = alerts_response.json()["data"]["alerts"]
     log.info(f"Prometheus Alerts: {alerts}")
-    assert constants.ALERT_PROMETHEUSRULEFAILURES not in [
-        alert["labels"]["alertname"] for alert in alerts
-    ]
+    test_results[f"{constants.ALERT_PROMETHEUSRULEFAILURES}-present-check"] = (
+        constants.ALERT_PROMETHEUSRULEFAILURES
+        not in [alert["labels"]["alertname"] for alert in alerts]
+    )
+    prometheus_pods = get_pod_name_by_pattern(
+        "prometheus-k8s", constants.MONITORING_NAMESPACE
+    )
+    for pod_name in prometheus_pods:
+        log.info(f"Checking logs of pod {pod_name}")
+        pod_logs = get_pod_logs(
+            pod_name=pod_name,
+            namespace=constants.MONITORING_NAMESPACE,
+        ).splitlines()
+        pod_logs.reverse()
+        for log_line in pod_logs:
+            if "many-to-many matching not allowed" in log_line.lower():
+                log.error(
+                    f"Found 'warn' msg in logs of pod {pod_name}:"
```

Comment:
IIUC, the fix has already been backported to a few zstreams and is pending for the others. Once the fix is fully backported to all relevant zstream releases, there is no need for this code?

Reply:
I think we want to cover this scenario to not miss bug in future.
This issue can return when alerts are modified or added, and warn message will be logged again.